server/sublist.go

```diff
@@ -266,7 +261,9 @@ func addNodeToResults(n *node, results *SublistResult) {
 		if i := findQSliceForSub(qr[0], results.qsubs); i >= 0 {
 			results.qsubs[i] = append(results.qsubs[i], qr...)
 		} else {
-			results.qsubs = append(results.qsubs, qr)
+			copyqr := make([]*subscription, len(qr))
+			copy(copyqr, qr)
+			results.qsubs = append(results.qsubs, copyqr)
```

Comment:
Any noticeable performance hit? Any other way to avoid the allocation and copy even of the array?